### DISCUSSION TITLE
docker build error

### DISCUSSION THREAD

**Discussion by ghost** [OrchardCMS/OrchardCore#16366] [Q&A]
In order to meet security requirements (45 day password expire), I need to change the users module (login function).   When trying to edit and build with source code I get the error below.  

Also, is editing the source the right way to make this change?  I tried creating an altered users module over the weekend, but had trouble when trying to replace the built in users module with it.  

PS C:\Repos\Orchard> docker compose up --build
2024/06/24 11:32:47 http2: server: error reading preface from client //./pipe/dockerDesktopLinuxEngine: file has already been closed
[+] Building 15.2s (9/9) FINISHED                                                                                                                                                    docker:desktop-linux
 => [fisorchard internal] load build definition from Dockerfile                                                                                                                                      0.0s
 => => transferring dockerfile: 371B                                                                                                                                                                 0.0s
 => [fisorchard internal] load metadata for mcr.microsoft.com/dotnet/sdk:8.0                                                                                                                         0.9s
 => [fisorchard internal] load .dockerignore                                                                                                                                                         0.1s
 => => transferring context: 291B                                                                                                                                                                    0.0s
 => [fisorchard build-env 1/5] FROM mcr.microsoft.com/dotnet/sdk:8.0@sha256:3189e564f19e016a43838a46609fc81349f07322fdf6bc3299bd13f0dca9e647                                                         0.0s
 => [fisorchard internal] load build context                                                                                                                                                         1.8s
 => => transferring context: 1.12MB                                                                                                                                                                  1.7s
 => CACHED [fisorchard build-env 2/5] WORKDIR /app                                                                                                                                                   0.0s
 => CACHED [fisorchard build-env 3/5] COPY ./src /app                                                                                                                                                0.0s
 => CACHED [fisorchard build-env 4/5] COPY Directory.Build.props /                                                                                                                                   0.0s
 => ERROR [fisorchard build-env 5/5] RUN dotnet publish /app/OrchardCore.Cms.Web -c Release -o ./build/release --framework net8.0 /p:RunAnalyzers=false                                             12.3s
------
 > [fisorchard build-env 5/5] RUN dotnet publish /app/OrchardCore.Cms.Web -c Release -o ./build/release --framework net8.0 /p:RunAnalyzers=false:
1.384   Determining projects to restore...
11.73 /usr/share/dotnet/sdk/8.0.302/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(96,5): error NETSDK1013: The TargetFramework value '' was not recognized. It may be misspelled. If not, then the TargetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly. [/app/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
------
failed to solve: process "/bin/sh -c dotnet publish /app/OrchardCore.Cms.Web -c Release -o ./build/release --framework net8.0 /p:RunAnalyzers=false" did not complete successfully: exit code: 1

**Comment by ghost** [comment]
Update:  these are the latest errors.  (this is using the latest source)
What would be the be way to tackle this?


#11 54.15 /src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/OrchardCore.Media.AmazonS3.csproj : error NU1107: Version conflict detected for AWSSDK.Core. Install/reference AWSSDK.Core 3.7.8.10 directly to project OrchardCore.Media.AmazonS3 to resolve this issue.  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 54.15 /src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/OrchardCore.Media.AmazonS3.csproj : error NU1107:  OrchardCore.Media.AmazonS3 -> SixLabors.ImageSharp.Web.Providers.AWS 2.0.0 -> AWSSDK.S3 3.7.8.8 -> AWSSDK.Core (>= 3.7.8.10 && < 4.0.0)  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 54.15 /src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/OrchardCore.Media.AmazonS3.csproj : error NU1107:  OrchardCore.Media.AmazonS3 -> OrchardCore.FileStorage.AmazonS3 -> AWSSDK.Extensions.NETCore.Setup 3.3.0 -> AWSSDK.Core (>= 3.3.0 && < 3.4.0). [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]

#11 58.33 /src/OrchardCore/OrchardCore.Search.Lucene.Core/OrchardCore.Search.Lucene.Core.csproj : error NU1103: Unable to find a stable package Lucene.Net.QueryParser with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 58.33 /src/OrchardCore/OrchardCore.Search.Lucene.Core/OrchardCore.Search.Lucene.Core.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 58.34 /src/OrchardCore/OrchardCore.Search.Lucene.Core/OrchardCore.Search.Lucene.Core.csproj : error NU1103: Unable to find a stable package Lucene.Net.Spatial with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 58.34 /src/OrchardCore/OrchardCore.Search.Lucene.Core/OrchardCore.Search.Lucene.Core.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]

#11 65.89 /src/OrchardCore/OrchardCore.FileStorage.AmazonS3/OrchardCore.FileStorage.AmazonS3.csproj : error NU1107: Version conflict detected for AWSSDK.Core. Install/reference AWSSDK.Core 3.3.0 directly to project OrchardCore.FileStorage.AmazonS3 to resolve this issue.  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 65.89 /src/OrchardCore/OrchardCore.FileStorage.AmazonS3/OrchardCore.FileStorage.AmazonS3.csproj : error NU1107:  OrchardCore.FileStorage.AmazonS3 -> AWSSDK.Extensions.NETCore.Setup 3.3.0 -> AWSSDK.Core (>= 3.3.0 && < 3.4.0)  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 65.89 /src/OrchardCore/OrchardCore.FileStorage.AmazonS3/OrchardCore.FileStorage.AmazonS3.csproj : error NU1107:  OrchardCore.FileStorage.AmazonS3 -> AWSSDK.S3 3.1.0 -> AWSSDK.Core (>= 3.1.0 && < 3.2.0). [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]

#11 69.87 /src/OrchardCore.Modules/OrchardCore.Search.Lucene/OrchardCore.Search.Lucene.csproj : error NU1103: Unable to find a stable package Lucene.Net.QueryParser with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 69.87 /src/OrchardCore.Modules/OrchardCore.Search.Lucene/OrchardCore.Search.Lucene.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 69.87 /src/OrchardCore.Modules/OrchardCore.Search.Lucene/OrchardCore.Search.Lucene.csproj : error NU1103: Unable to find a stable package Lucene.Net.Spatial with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 69.87 /src/OrchardCore.Modules/OrchardCore.Search.Lucene/OrchardCore.Search.Lucene.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]

#11 75.32 /src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1103: Unable to find a stable package Lucene.Net.QueryParser with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 75.32 /src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 75.32 /src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1103: Unable to find a stable package Lucene.Net.Spatial with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 75.32 /src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]

#11 75.42 /src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1107: Version conflict detected for AWSSDK.Core. Install/reference AWSSDK.Core 3.7.8.10 directly to project OrchardCore.Application.Cms.Core.Targets to resolve this issue.  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 75.42 /src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1107:  OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> SixLabors.ImageSharp.Web.Providers.AWS 2.0.0 -> AWSSDK.S3 3.7.8.8 -> AWSSDK.Core (>= 3.7.8.10 && < 4.0.0)  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 75.42 /src/OrchardCore/OrchardCore.Application.Cms.Core.Targets/OrchardCore.Application.Cms.Core.Targets.csproj : error NU1107:  OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> OrchardCore.FileStorage.AmazonS3 -> AWSSDK.Extensions.NETCore.Setup 3.3.0 -> AWSSDK.Core (>= 3.3.0 && < 3.4.0). [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]

#11 76.53 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1103: Unable to find a stable package Lucene.Net.QueryParser with version 
#11 76.53 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ]
#11 76.53 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1103: Unable to find a stable package Lucene.Net.Spatial with version 
#11 76.53 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ]

#11 76.53 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1103: Unable to find a stable package Lucene.Net.QueryParser with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 76.53 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 76.53 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1103: Unable to find a stable package Lucene.Net.Spatial with version  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 76.53 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1103:   - Found 14 version(s) in nuget.org [ Nearest version: 4.8.0-beta00016 ] [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]

#11 76.59 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1107: Version conflict detected for AWSSDK.Core. Install/reference AWSSDK.Core 3.7.8.10 directly to project OrchardCore.Application.Cms.Targets to resolve this issue.  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 76.59 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1107:  OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> SixLabors.ImageSharp.Web.Providers.AWS 2.0.0 -> AWSSDK.S3 3.7.8.8 -> AWSSDK.Core (>= 3.7.8.10 && < 4.0.0)  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 76.59 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1107:  OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> OrchardCore.FileStorage.AmazonS3 -> AWSSDK.Extensions.NETCore.Setup 3.3.0 -> AWSSDK.Core (>= 3.3.0 && < 3.4.0). [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
#11 76.59 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1107: Version conflict detected for AWSSDK.Core. Install/reference AWSSDK.Core 3.7.8.10 directly to project OrchardCore.Cms.Web to resolve this issue. 
#11 76.59 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1107:  OrchardCore.Cms.Web -> OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> SixLabors.ImageSharp.Web.Providers.AWS 2.0.0 -> AWSSDK.S3 3.7.8.8 -> AWSSDK.Core (>= 3.7.8.10 && < 4.0.0) 
#11 76.59 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1107:  OrchardCore.Cms.Web -> OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> OrchardCore.FileStorage.AmazonS3 -> AWSSDK.Extensions.NETCore.Setup 3.3.0 -> AWSSDK.Core (>= 3.3.0 && < 3.4.0).

76.59 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1107: Version conflict detected for AWSSDK.Core. Install/reference AWSSDK.Core 3.7.8.10 directly to project OrchardCore.Application.Cms.Targets to resolve this issue.  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
76.59 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1107:  OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> SixLabors.ImageSharp.Web.Providers.AWS 2.0.0 -> AWSSDK.S3 3.7.8.8 -> AWSSDK.Core (>= 3.7.8.10 && < 4.0.0)  [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
76.59 /src/OrchardCore/OrchardCore.Application.Cms.Targets/OrchardCore.Application.Cms.Targets.csproj : error NU1107:  OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> OrchardCore.FileStorage.AmazonS3 -> AWSSDK.Extensions.NETCore.Setup 3.3.0 -> AWSSDK.Core (>= 3.3.0 && < 3.4.0). [/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj]
76.59 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1107: Version conflict detected for AWSSDK.Core. Install/reference AWSSDK.Core 3.7.8.10 directly to project OrchardCore.Cms.Web to resolve this issue. 
76.59 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1107:  OrchardCore.Cms.Web -> OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> SixLabors.ImageSharp.Web.Providers.AWS 2.0.0 -> AWSSDK.S3 3.7.8.8 -> AWSSDK.Core (>= 3.7.8.10 && < 4.0.0) 
76.59 /src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj : error NU1107:  OrchardCore.Cms.Web -> OrchardCore.Application.Cms.Targets -> OrchardCore.Application.Cms.Core.Targets -> OrchardCore.Media.AmazonS3 -> OrchardCore.FileStorage.AmazonS3 -> AWSSDK.Extensions.NETCore.Setup 3.3.0 -> AWSSDK.Core (>= 3.3.0 && < 3.4.0).

**Reply by sloethen** [reply]
I am seeing same line fail when running docker build.  it is reporting an empty framework parameter.
